src/esy.ml

```diff
@@ -92,6 +92,20 @@ module State = struct
     | Pending
 end
 
+let find_manifest_in_dir dir =
+  let open Promise.Syntax in
+  let esy_file = Path.(dir / "esy.json") in
+  let package_file = Path.(dir / "package.json") in
+  try
+    [ esy_file; package_file ]
+    |> Promise.List.find_map (fun path ->
+           let+ file_exists = path |> Path.to_string |> Fs.exists in
+           if file_exists then
+             Some path
+           else
+             None)
+  with Assert_failure _ -> Promise.return None
```

Comment:
Rudi said something about this exception earlier; ignore if resolved

Reply:
Ah my bad: I saw @mnxn pushed the upgrade of `jsoo_promise` and assumed he did it at the same time. Will update.